### PR TITLE
Closes #2339 split `derive_date_vars.R` file into inherent component parts

### LINKED ISSUE #2339: Separate `derive_date_vars()` into its two inherently separate compoents: datetime vs date and their respective parent-child subfunctions
`derive_date_vars.R` breaks the convention of how we usually format our files, it is difficult to navigate, would be better if it was broken down into its inherent two component parts: `derive_vars_dt()` and `derive_vars_dtm()`

**Defintion of done:**

1. Split `derive_date_vars,R` into `derive_vars_dt.R`, `derive_vars_dtm.R` `date_vars_utils.R`
2. `derive_vars_dt.R` would contain `derive_vars_dt()` `convert_dtc_to_dt()` `impute_dtc_dt()` `restricted_imputed_dtc_dt()` and `compute_dtf()`
3. `derive_vars_dtm.R` would contain `derive_vars_dtm()`  `get_partial_datetime()` `convert_dtc_to_dtm()` `impute_dtc_dtm()` `restricted_imputed_dtc_dtm()` and `compute_tmf()` 
4. `date_vars_utils.R` would contain `dtm_level()`  `dt_level()` `get_imputation_target_date()`  `get_imputation_target_time()`  `convert_date_to_dtm()`

### PR DESCRIPTION
**Pull request by zdz2101**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/CONTRIBUTING.html#detailed-development-process) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `main` branch until you have checked off each task.

- [ ] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [ ] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [ ] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [ ] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#deprecation)?
- [ ] Review the [Cheat Sheet](https://github.com/pharmaverse/admiral/blob/main/inst/cheatsheet/admiral_cheatsheet.pdf). Make any required updates to it by editing the file `inst/cheatsheet/admiral_cheatsheet.pptx` and re-upload a PDF version of it to the same folder.
- [ ] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [ ] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [ ] Address any updates needed for vignettes and/or templates
- [ ] Update `NEWS.md` under the header `# admiral (development version)` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [ ] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/reference/index.html)" page. 
- [ ] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [ ] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [ ] Link the issue in the Development Section on the right hand side.
- [ ] Address all merge conflicts and resolve appropriately
- [ ] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Review by bms63 — COMMENTED** [review]
Why have so many files `.Rd` files changed?

File: man/compute_dtf.Rd [hunk 1]
```diff
@@ -1,5 +1,5 @@
 % Generated by roxygen2: do not edit by hand
-% Please edit documentation in R/derive_date_vars.R
+% Please edit documentation in R/derive_vars_dt.R
```
Comment (zdz2101) [inline man/compute_dtf.Rd:2]: > Why have so many files `.Rd` files changed?

The migration caused at least these changes, as for the others I'm not really sure, just ran `devtool::document()` like usual

**Review by zdz2101 — COMMENTED** [review]
See comment @bms63

**Comment by bms63** [comment]
The roxygen test is failing.  Are you using the same roxygen package in the CI?

**Comment by zdz2101** [comment]
That was definitely it, good catch, the still-many .Rd files pertain to the migration

**Comment by millerg23** [comment]
@zdz2101 this looks great, only question I have is, why did you put `get_partialdatetime` in `derive_vars_dtm.R` file, instead of `date_vars_util.R` ?

**Comment by zdz2101** [comment]
> @zdz2101 this looks great, only question I have is, why did you put `get_partialdatetime` in `derive_vars_dtm.R` file, instead of `date_vars_util.R` ?

Oddly enough, this function is only used for the `derive_vars_dtm()` , the naming of it certainly makes more sense/has a feel of a utils though I know what you mean, I'm okay with moving it there!

**Comment by bms63** [comment]
Shall we just do the tests split in this PR as well? #2340.

The number of issues is starting to freak me out again!!

image

**Comment by millerg23** [comment]
> Shall we just do the tests split in this PR as well? #2340.

Yes I thnk that make sense, we can then tackle each of the split out unit test files to check coverage.

**Pull request merged**